app/services/BusinessDetailsService.scala

```diff
@@ -16,20 +16,51 @@
 
 package services
 
+import config.MicroserviceAppConfig
 import connectors.BusinessDetailsConnector
-import models.incomeSourceDetails.{Nino, IncomeSourceDetailsResponseModel}
+import connectors.hip.GetBusinessDetailsConnector
+import models.hip.GetBusinessDetailsHipApi
 import play.api.Logging
+import play.api.http.Status.OK
+import play.api.libs.json.Json
+import play.api.mvc.Result
+import play.api.mvc.Results.Status
 import uk.gov.hmrc.http.HeaderCarrier
 
 import javax.inject.{Inject, Singleton}
-import scala.concurrent.Future
+import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
-class BusinessDetailsService @Inject()(val businessDetailsConnector: BusinessDetailsConnector) extends Logging {
+class BusinessDetailsService @Inject()(val businessDetailsConnector: BusinessDetailsConnector,
+                                       val getBusinessDetailsFromHipConnector: GetBusinessDetailsConnector,
+                                       val appConfig: MicroserviceAppConfig
+                                      ) extends Logging {
 
-  def getBusinessDetails(nino: String)(implicit headerCarrier: HeaderCarrier): Future[IncomeSourceDetailsResponseModel] = {
+  def getBusinessDetails(nino: String)
+                        (implicit headerCarrier: HeaderCarrier,
+                         ec:ExecutionContext): Future[Result] = {
     logger.debug("Requesting Income Source Details from Connector")
-    businessDetailsConnector.getBusinessDetails(nino, Nino)
+    if(appConfig.hipFeatureSwitchEnabled(GetBusinessDetailsHipApi))
+      getBusinessDetailsFromHipConnector.getBusinessDetails(nino, models.hip.incomeSourceDetails.Nino).map {
+        case success: models.hip.incomeSourceDetails.IncomeSourceDetailsModel =>
```

Comment:
wondering but can the error and success models be changed to have a Hip prefix in the naming? 
e.g. 
 HipIncomeSourceDetailsModel or 
 HipIncomeSourceDetailsNotFound or
 HipIncomeSourceDetailsResponseModel etc. 
 
 think then the import can be just pulled in and it's easier to differentiate between the two.

Reply:
Could do. But only thought its easier to just delete old IncomeSource models when we cleanup.